extends Node2D

@onready var network_manager: Node = $Network_Manager
var player_scene = preload("res://scenes/actors/player/player.tscn")
var players = {}

func _ready():
	if "--server" in OS.get_cmdline_args():
		iniciar_servidor()
	else:
		print("ğŸŸ¡ Rodando como cliente, aguardando conexÃ£o do menu...")

func iniciar_servidor():
	print("ğŸ–¥ï¸ Servidor iniciado.")
	network_manager.start_server()  # Certifique-se de que esta funÃ§Ã£o existe no NetworkManager

func connect_to_server(ip):
	print("ğŸ”Œ Tentando conectar ao servidor em ", ip)
	network_manager.connect_to_server(ip)  # FunÃ§Ã£o que conecta ao servidor

func _on_player_connected(id):
	print("ğŸ‘¤ Novo jogador conectado! ID: ", id)  # Verifica se o jogador estÃ¡ sendo criado
	spawn_player(id)

func spawn_player(id):
	print("ğŸ“Œ Tentando spawnar jogador ID: ", id)
	if id in players:
		return
	
	var player = player_scene.instantiate()
	player.name = str(id) #nomeamos o nÃ³ com o ID do jogador
	add_child(player)
	players[id] = player
	
	if multiplayer.is_server():
		player.position = Vector2(400, 300) #define a posiÃ§Ã£o inicial para novos jogadores 

func send_player_name():
	if multiplayer.is_server():
		print("ğŸ–¥ï¸ Servidor iniciado. Nenhum nome serÃ¡ enviado.") 
		return
	
	var nome = "Jogador " + str(randi() % 1000)
	print("ğŸ“¨ Enviando nome:", nome)
	network_manager.set_player_name.rpc_id(1, multiplayer.get_unique_id(), nome)
